tests/queries/0_stateless/02354_with_statement_non_exist_column.sql

```diff
@@ -0,0 +1 @@
+WITH x AS y SELECT 1;
```

Comment:
So, as I get from the liked issue we need to support it because of backward compatibility? We didn't throw error even if `x` is really doesn't exists in some previous versions, that's why we want to support it now?

Reply:
I don't understand that test. It's something strange, and i don't know why it should work. :) 

In the original report it was real columns which existsed.

@amosbird maybe it's worth to add the test from the initial report?

Reply:
I mean that https://github.com/ClickHouse/ClickHouse/issues/37812#issue-1259568892

Reply:
Probably we should decide that this one should not work and the correct one should?

Reply:
The test case is properly showing the root cause.

Here is the original case:

```
CREATE TEMPORARY TABLE t1 (a Int64);
CREATE TEMPORARY TABLE t2 (a Int64, b Int64);

-- works
WITH b AS bb SELECT bb FROM t2 WHERE a IN t1;     

-- fails
WITH b AS bb SELECT bb FROM t2 WHERE a IN (SELECT a FROM t1);
```

It doesn't work because `enable_global_with_statement = 1` pushes down the with statement into `SELECT a FROM t1` and table `t1` doesn't have column `b` .

Reply:
Couldn't we fix original cause with push down? Just ignoring unknown unused columns doesn't seems right, I suppose it's fine to have an error in query from test.

Reply:
During syntax analysis we don't know the required columns needed by subqueries. I don't think there is a way to only push down needed `with aliases `.

Reply:
It can also become complex when subquery contains another subquery which needs the unused column.

Reply:
I think the ideal solution here is to have different scopes or variables / columns. 
Child can access parents' WITH scope without copying it to it's own scope. 
Things from the current scope should 'hide' the things from the 'parent' scope without 

But for now, i think that fix is ok.

Reply:
> I think the ideal solution here is to have different scopes or variables / columns.

Yeah, but we don't have such infrastructure for now)